aries-sdk-core/src/main/java/com/hyperledger/aries/messaging/AgentMessage.java

```diff
@@ -0,0 +1,43 @@
+package com.hyperledger.aries.messaging;
+
+import com.google.gson.Gson;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.Optional;
+
+public class AgentMessage {
+    @SerializedName("@type")
+    private String type;
+
+    @SerializedName("@id")
+    private String id;
+
+    AgentMessage(String type, String id) {
+        this.type = type;
+        this.id = id;
+    }
+
+    public static Optional<AgentMessage> from(String rawJson) {
+        try {
+            AgentMessage message = new Gson().fromJson(rawJson, AgentMessage.class);
+            if (null == message.getType() || null == message.getId()) {
```

Comment:
You can also think of transforming this blocj like ofNullable(message.getType()).orElseThrow()

Reply:
- orEleseThrow() was applied
- Added 2 unit-tests (null type and null id)